test/cases/coerced_tests.rb

```diff
@@ -47,10 +47,17 @@ def test_value_limit_violations_are_translated_to_specific_exception_coerced
 module ActiveRecord
   class AdapterPreventWritesTest < ActiveRecord::TestCase
     # Fix randomly failing test. The loading of the model's schema was affecting the test.
+    # Also, we do some read queries. Remove assert_no_queries
```

Comment:
What kind read queries we do? The fact that we are coercing tests like this make me fell we are missing some `SCHEMA` tags when querying for internal information.

Reply:
I don't see this test failing in the latest CI build (https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/runs/2399437373). The issue with unexpected read queries was fixed in the past by `Subscriber.send(:load_schema!)` so I don't know why it would be no longer working. 

I think the changes to `test_errors_when_an_insert_query_is_called_while_preventing_writes_coerced` should be removed and if the issue happens again on the CI we can look at it again.

Reply:
Mmmm, but I do see the test failing in some runs https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/runs/2400618957?check_suite_focus=true#step:4:440 (my last commit run).

I checked the last run and you were right, it wasn't failing. But there was another one failing https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/runs/2399437340?check_suite_focus=true#step:4:695. I coerced that test following your suggestion about using `Subscriber.send(:load_schema!)`.

In some ruby versions the test is still failing https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/runs/2400618873?check_suite_focus=true#step:4:448

And in other ruby version the original (the one I changed) is failing https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/runs/2400618957?check_suite_focus=true#step:4:440.

I can't explain this.